### PR TITLE
Fix SHA256, add macOS, add activate.sh

### PR DESCRIPTION
**Pull request by tyhoff**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Unsure why the SHA256 was incorrect from the bot, but it was!

While I'm here and at it, improve the recipe with improvements I made to my own personal version of cpputest: https://github.com/memfault/conda-recipes/tree/master/cpputest

- Add macOS support
- Add `(de)activate.sh` scripts to set `CPPUTEST_HOME` so that the library is automatically found by the internal scripts
- Add myself as a maintainer

I hope to backport these improvements to CppUTest v3.8 as well.

### CODE REVIEW

**Comment by tyhoff** [comment]
@conda-forge-admin, please rerender

**Comment by tyhoff** [comment]
@conda-forge/core @avrahamshukron ready for review!

File: recipe/activate.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# CppUTest will look for this variable to find the library
+# https://cpputest.github.io/manual.html
+export CPPUTEST_HOME="$CONDA_PREFIX"
```
Comment (avrahamshukron) [inline recipe/activate.sh]:
```suggestion
export CPPUTEST_HOME="$CONDA_PREFIX"

```

Newline at the end of file :)

File: recipe/deactivate.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+unset CPPUTEST_HOME
```
Comment (avrahamshukron) [inline recipe/deactivate.sh]:
```suggestion
unset CPPUTEST_HOME

```
Newline

**Review by avrahamshukron — APPROVED** [review]
Looks good!

**Pull request merged**